base/abstractarray.jl

```diff
@@ -2229,7 +2230,14 @@ end
 # map on collections
 map(f, A::AbstractArray) = collect_similar(A, Generator(f,A))
 
-mapany(f, itr) = map!(f, Vector{Any}(undef, length(itr)::Int), itr)  # convenient for Expr.args
+mapany(f, A::AbstractArray) = map!(f, Vector{Any}(undef, length(A)), A)
+function mapany(f, itr)
+    ret = []
+    for x in itr
+        push!(ret, f(x))
+    end
+    return ret
+end
```

Comment:
Maybe,
```suggestion
mapany(f, itr) = Any[f(x) for x in itr]
```
?

Reply:
Yes, that seems safe. I wasn't certain originally but I've verified that never calls `similar`.